go/pkg/yolostore/gorm.go

```diff
@@ -0,0 +1,135 @@
+	db = db.Set("gorm:auto_preload", false)
+	db = db.Set("gorm:association_autoupdate", false)
+	db.BlockGlobalUpdate(true)
+	db.SingularTable(true)
+	db.LogMode(true)
+	if err := db.AutoMigrate(yolopb.AllModels()...).Error; err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+func beforeCreate(scope *gorm.Scope) {
+	if !scope.HasColumn("yolo_id") {
+		return
+	}
+
+	field, found := scope.FieldByName("id")
+	if !found {
+		return
+	}
+
+	table := scope.TableName()
+	id := field.Field.String()
+	hash := sha256.Sum256([]byte(id))
+	encoded := base58.Encode(hash[:])
+	prefix := strings.ToLower(table[:1])
+	yoloID := fmt.Sprintf("%s:%s", prefix, encoded)
+	err := scope.SetColumn("yolo_id", yoloID)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func TestingDB(t *testing.T) *gorm.DB {
+	db, err := gorm.Open("sqlite3", "file::memory:?cache=shared")
+	if err != nil {
+		t.Fatalf("init in-memory sqlite server: %v", err)
+	}
+
+	logger, _ := zap.NewProduction()
+
+	db, err = initDB(db, logger)
+	if err != nil {
+		t.Fatalf("init in-memory db: %v", err)
+	}
+
+	TestingCreateEntities(t, db)
+
+	return db
+}
+
+func TestingCreateEntities(t *testing.T, db *gorm.DB) {
+	if err := db.Transaction(func(tx *gorm.DB) error {
+
+		// create artifact
+		var artifact *yolopb.Artifact
+		artifact = &yolopb.Artifact{
+			ID:          "artif1",
+			FileSize:    80,
+			LocalPath:   "js/packages/bla",
+			DownloadURL: "https://api.buildkite.com",
+			MimeType:    "application/octet-stream",
+			State:       1,
+			Kind:        2,
+			Driver:      1,
+			HasBuildID:  "https://buildkite.com/berty/berty/builds/2738",
+		}
+		if err := db.Create(artifact).Error; err != nil {
+			t.Fatalf("create artifact: %v", err)
+		}
+		// create build
+		var build *yolopb.Build
+		build = &yolopb.Build{
+			ID:                "https://buildkite.com/berty/berty/builds/2738",
+			State:             1,
+			Message:           "feat: tests",
+			Branch:            "feat/tests",
+			Driver:            1,
+			ShortID:           "1000",
+			HasCommitID:       "commit1",
+			HasProjectID:      "https://github.com/berty/berty",
+			HasMergerequestID: "https://github.com/berty/berty/pull/2438",
+		}
+		if err := db.Create(build).Error; err != nil {
+			t.Fatalf("create build: %v", err)
+		}
+		var entity *yolopb.Entity
+		entity = &yolopb.Entity{
+			ID:          "entity1",
+			Name:        "entity1",
+			Driver:      1,
+			AvatarURL:   "url",
+			Kind:        1,
+			Description: "description",
+		}
+		if err := db.Create(entity).Error; err != nil {
+			t.Fatalf("create entity: %v", err)
+		}
+		var project *yolopb.Project
+		project = &yolopb.Project{
+			ID:          "proj1",
+			Driver:      1,
+			Name:        "proj1",
+			Description: "description",
+			HasOwnerID:  "1",
+		}
+		if err := db.Create(project).Error; err != nil {
+			t.Fatalf("create entity: %v", err)
+		}
+		return nil
+	}); err != nil {
+		t.Fatalf("create testing entities: %v", err)
```

Comment:
since you're using a final `t.Fatal`, you should just use standard `return fmt.Errorf("blah: %w")` inside the transaction